### PR TITLE
#168107627 Add check for type of params passed into url

### PR DESCRIPTION
**Pull request by RonKbS**
#### What does this PR do?
- Adds check for the type of parameters passed into a URL that can return filtered results.

#### Description of Task to be completed?
-  add the check in the security.py file
- Adjust existing tests to ensure they pass

#### How should this be manually tested?
- Pull and checkout to this branch locally by running `git fetch` && git checkout `ch-check-query-param-type-168107627`

- Run the server and go to the URL `http://127.0.0.1:5000/api/v1/menu_template_items?day_id=<id>` and pass in an invalid id.

- Run tests `pytest -k pytest -k test_security`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#168107627](https://www.pivotaltracker.com/story/show/168107627)

### CODE REVIEW

File: tests/unit/utils/test_security.py [hunk 1]
```diff
@@ -424,8 +424,28 @@ class MockRequest:
...
             response[0].get_json()['msg'],
-            "Invalid keys ['unknown']. The supported keys are "
-            "['Id', 'IsDeleted', 'CreatedAt', 'UpdatedAt', 'Category', 'Question', 'Answer']"
+            "Invalid parameters ['unknown']. The supported keys and value-types are "
+            "['Id', 'IsDeleted', 'CreatedAt', 'UpdatedAt', 'Category', 'Question', "
+            "'Answer', 'BOOLEAN', 'DATETIME', 'INTEGER', 'VARCHAR(2000)', 'VARCHAR(9)']"
+        )
+
+
+    def test_validate_query_params_validates_model_field_types(self):
```
Comment (kimbugp) [inline tests/unit/utils/test_security.py]: Nice work @RonKbS,
I suggest you add some end to end tests for this implementation

File: app/utils/security.py [hunk 2]
```diff
@@ -380,15 +381,28 @@ def validator(f):
...
+                        continue
+
+                invalid_params = invalid_query_keys + invalid_query_values
+                listed_mfd = list(model_field_types)
+                listed_mfd.sort()
+                valid_params = model_fields_camel + listed_mfd
+
+                if invalid_query_keys or invalid_query_values:
                     return make_response(
-                        jsonify({'msg': 'Invalid keys {}. The supported keys are {}'
```
Comment (kimbugp) [inline app/utils/security.py]: @RonKbS, I suggest you also don't expose the data types in our db such as varchar, boolean
I think a better way would be to create a more meaning full error message

Comment (RonKbS) [reply app/utils/security.py]: This has been worked on @kimbugp.

File: tests/unit/utils/test_security.py (hunk 1 above)
Comment (RonKbS) [reply tests/unit/utils/test_security.py]: An additional end-to-end test has been added.

**Pull request merged**